### ISSUE TITLE
ForwardToConsole=yes doesn't work on VirtualBox or KVM machines

### ISSUE THREAD

**Issue by q2dg** [systemd/systemd#10703]
**systemd version the issue has been seen with**
> 239

**Used distribution**
 > Fedora 29

**Expected behaviour you didn't see**
 > If I edit /etc/systemd/journald.conf on a real bare-metal Fedora 29 system (Server or Workstation, I've tried both), writting ForwardToConsole=yes, TTYPath=/dev/tty3 (or TTYPath=/dev/pts/1, it's the same) and MaxLevelConsole=debug I can see messages on specified console like a charm (trying, for example, with _echo hello | systemd-cat -t "tag" -p err_).

But if I do exactly the same inside a virtual Fedora 29 system (Server or Workstation) using either VirtualBox or Qemu/KVM/Virt-manager, it doesn't work (!!??). I mean: messages aren't shown on specified console. 

I think it could be a rare bug but maybe there is a reasonable (though obscure) explanation
Thanks a lot.

**Comment by poettering** [comment]
what makes you think systemd is at fault here? if you run "hexdump -C /dev/vcsa3", do you see the output generated?

Also, /dev/tty3 and /dev/pts/1 is definitely *not* the same... The former is a virtual console, the latter a pseudo console. That's very different things on Linux.

Did you restart journald after making that change?

**Comment by q2dg** [comment]
Yes, I've restarted journald after making that change.
On the other hand...I explained myself very badly, when I said /dev/tty3 and /dev/pts/1 were the same I wanted to say that redirecting to /dev/tty3 in Fedora Server and redirecting to /dev/pts/1 in Fedora Workstation has the same behaviour: messages aren't shown.

Well, I've discoverd something new, though. If in /dev/tty3 there isn't any user session started, it works. It doesn't work when there is an user session already started, whatever it is. Obviously, sending messages to /dev/pts/1 never works because there's always the desktop user session started. So I suspect the problem is related to this.
Thanks a lot for your time

**Comment by poettering** [comment]
/dev/pts/1 needs to be allocated by some tool before it is around, and the tool needs to stick around. If you want that journald logs to it then you'd need to allocate this before journald even starts... hence what you are trying to do makes not much sense to me. what tool is allocated that pseudo tty for you?

**Comment by poettering** [comment]
also, what does "aren#t shown" mean? did you switch to VC 3 when using /dev/tty3?

**Comment by q2dg** [comment]
Hello.

I can confirm _hexdump -C /dev/vsc3_ doesn't show anything if journald messages are redirected to a tty (/dev/tty3) and if there is an active user session started in that tty. But the strange thing is this only happens in a VM machine :-/

On the other hand, in Fedora Workstation I use gnome-terminal to get into /dev/pts/1 inside a regular Gnome session. I just wanted to show to my students  the easiness of doing the redirection of journal messages in a "visual" way. I agree it's not a real useful scenario (so maybe it's not very important case) but what surprises me a lot is that this redirection works on a real Fedora system but not on a virtual one.

Thanks

**Comment by q2dg** [comment]
Sorry, I've finally found the error. I didn't remember that I disabled SELinux in my real Fedora system many time ago but I am working with VMs running "stock" Fedora systems (that is, with SELinux activated). So doing a _setenforce permissive_ in any VM solves the issue.

In fact, working with my Fedora VMs I had many other issues I haven't explained here, like being unable to execute the _machinectl start_ command or the _machinectl login_ command, and so on, but all of this was due to the same: SELinux. I was obtaining very strange errors of "operation not permitted" that lead me to this key comment: https://github.com/systemd/systemd/issues/1446#issuecomment-144999339

Thanks all of you for your patiente and I apologize for the inconvenience.
